purple-pill/js/controllers/fontexportController.js

```diff
@@ -20,152 +92,136 @@ function($scope, $http, sharedScope, $timeout) {
-        resetProgressBar();
-
-        function zero_padding(value){
+    function getTimestamp(){
+        var year, month, day, hours, minutes, seconds
+          , date = new Date()
+          ;
+        function zeroPadding(value){
             return value < 10 ? "0" + String(value) : String(value);
         }
+        year = zeroPadding(date.getFullYear());
+        month = zeroPadding(date.getMonth() + 1); //getMonth() returns a zero-based value: 0=January, 11=December
+        day = zeroPadding(date.getDate());
+        hours = zeroPadding(date.getHours());
+        minutes = zeroPadding(date.getMinutes());
+        seconds = zeroPadding(date.getSeconds());
+
+        return [year, month, day].join("") + "-" + [hours, minutes, seconds].join("");
+    }
+
+    const UI_UPDATE_TIMESLICE = 50; // msecs
+    var exportIsRunning = false;
+
+    function exportingGlyphMessage (it, instanceIndex, totalInstances){
+        var msg
+          , currentGlyph = it.value['current_glyph'] + 1 //humans start counting from 1.
+          , totalGlyphs = it.value['total_glyphs']
+          , glyphId = it.value['glyph_id']
+          ;
+        msg = totalInstances + " instances to export."
+        msg += "Calculating glyph '" + glyphId + "' (" + currentGlyph + " of " + totalGlyphs + ")"
+        msg += " of instance #" + (instanceIndex+1);
+        return msg;
+    }
+
+    function calculateGlyphsProgress(it, instanceIndex, totalInstances){
+        var percentage
+          , currentGlyph = it.value['current_glyph']
+          , totalGlyphs = it.value['total_glyphs']
+          ;
+        percentage = 100.0 * (instanceIndex + (currentGlyph/totalGlyphs)) / totalInstances;
+        return percentage;
+    }
 
-        function get_timestamp(){
-            var year, month, day, hours, minutes, seconds
-              , date = new Date()
-              ;
-            year = zero_padding(date.getFullYear());
-            month = zero_padding(date.getMonth() + 1); //getMonth() returns a zero-based value: 0=January, 11=December
-            day = zero_padding(date.getDate());
-            hours = zero_padding(date.getHours());
-            minutes = zero_padding(date.getMinutes());
-            seconds = zero_padding(date.getSeconds());
-
-            return [year, month, day].join("") + "-" + [hours, minutes, seconds].join("");
-        }
-
+    function exportFont_compute_glyphs(exportObjects) {
         var bundle = new $scope.data.stateless.JSZip()
-          , bundleFolderName = "metapolator-export-" + get_timestamp()
-          , bundle_filename = bundleFolderName + ".zip"
+          , bundleFolderName = "metapolator-export-" + getTimestamp()
+          , bundleFileName = bundleFolderName + ".zip"
           , bundleFolder = bundle.folder(bundleFolderName)
-          , bundle_data
-          , instances_for_export = Array()
+          , totalInstances = exportObjects.length
+          , progress = new ProgressBar( $("#progressbar")
```

Comment:
Don't create `progress` in here create it at the call to `exportFont_compute_glyphs` and inject it.